src/Components/WebAssembly/Server/src/PublicAPI.Unshipped.txt

```diff
@@ -1,4 +1,10 @@
 #nullable enable
+Microsoft.AspNetCore.Builder.WebAssemblyRazorComponentEndpointConventionBuilderExtensions
+Microsoft.AspNetCore.Components.Endpoints.WebAssemblyComponentsEndpointOptions
```

Comment:
This is another one where I personally prefer `WebAssemblyComponentsEndpointOptions`, but we approved `WebAssemblyComponentsEndpointsOptions` in [the API approval](https://github.com/dotnet/aspnetcore/issues/49754#issuecomment-1660809201). I think we can leave an API review note about going back to `Endpoint` if @SteveSandersonMS is fine with it.

I do think we should change the namespace though.

```suggestion
Microsoft.AspNetCore.Components.WebAssembly.Server.WebAssemblyComponentsEndpointOptions
```

Reply:
I'll go with whatever the two of you prefer.

Reply:
I also like `WebAssemblyComponentsEndpointOptions` (no "s"). Should we have a universal rules across these APIs that "Components" is always plural and "Endpoint" is always singular?

We might also have to change some other APIs if we decide that, though (e.g., RazorComponentsEndpoint~~s~~Options).

Reply:
I like this too. We could have called it `IEndpointsRouteBuilder` or `IEndpointsConventionBuilder` but we kept `EndpointSingular` there since it's an adjective.